options/options.go

```diff
@@ -52,10 +52,17 @@ type interpreter struct {
 	file                    file
 	resolver                linker.Resolver
 	overrideDescriptorProto linker.File
-	lenient                 bool
-	reporter                *reporter.Handler
-	index                   sourceinfo.OptionIndex
-	pathBuffer              []int32
+
+	index      sourceinfo.OptionIndex
+	pathBuffer []int32
+
+	reporter internal.ErrorHandler
+	lenient  bool
```

Comment:
I'm happy to rename some of these fields if you think that would help readability. The existing `lenient` flag means the interpreter will tolerate certain kinds of errors. The new flag below is set to true around the parts of the code that can generate those kinds of errors. (So errors can still occur in lenient mode -- it's just certain kind of errors that lenient mode tolerates.)